lib/cors.js

```diff
@@ -110,6 +109,12 @@ internals.handler = function (request, h) {
 
     // Validate Origin header
 
+    const origin = request.headers.origin;
+
+    if (!origin && !settings._ignoreOrigin) {
+        throw Boom.notFound('CORS error: Missing Origin header');
```

Comment:
Isn't this a violation of the CORS spec?

Reply:
> A resource that is publicly accessible, with no access control checks, can always safely return an `Access-Control-Allow-Origin` header whose value is "*".

Eg. it can send an invalid `origin: *` header, when the response would also set `access-control-allow-credentials: true`.

This can be guarded for in the Joi schema, by not allowing `wildcardIgnoresOrigin` to be used with `credentials`.

Reply:
I'm not sure that the spec forbids allowing credentials with a wildcard origin (I don't see anything in the spec pointing to it), but I think adding that restriction is a good security measure. The main use case for this flag is to improve caching of static files, and that still works if we add the restriction.

Should I add it then?